### PR TITLE
toolbar panes

### PR DESCRIPTION
**Pull request by yoshokatana**
* added `pane` service with `close` and `open` methods
* added `openPublish` method to open publish pane
* added `openValidationErrors` method to open validation errors pane
* added pane styles
* added tests

### CODE REVIEW

File: controllers/publish-pane.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    },
+
+    onPublishNow: function () {
+      return validation.validate(rules).then(function (errors) {
+        if (errors.length) {
+          pane.close();
+          pane.openValidationErrors(errors);
+        } else {
+          return edit.publishPage().then(function () {
+            alert('published!'); // eslint-disable-line
```
Comment (yoshokatana) [inline controllers/publish-pane.js]: once this is in, I'll add the progress bars and the status messages. Right now it's just an alert :smile:

File: services/pane.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+/**
+ * create pane
+ * @param {string} header
+ * @param {Element|DocumentFragment} innerEl
+ * @returns {Element}
+ */
+function createPane(header, innerEl) {
+  var template = dom.find('.kiln-pane-template'),
+    el = document.importNode(template.content, true);
```
Comment (yoshokatana) [inline services/pane.js]: html templates are neat

File: controllers/kiln-toolbar.js [hunk 3]
```diff
@@ -64,16 +64,7 @@ EditorToolbar.prototype = {
 
   onPublishClick: function openPublishPane() {
```
Comment (cruzanmo) [inline controllers/kiln-toolbar.js]: Why the wrapper function? vs. `onPublishClick: pane.openPublish`

File: services/pane.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  var pane  = dom.find('.kiln-toolbar-pane-background');
+
+  if (pane) {
+    dom.removeElement(pane);
+  }
+}
+
+/**
+ * open a pane
+ * @param {string} header
```
Comment (cruzanmo) [inline services/pane.js]: Could be helpful to have a short description for "header"

File: controllers/kiln-toolbar.js (hunk 3 above)
Comment (yoshokatana) [inline controllers/kiln-toolbar.js]: I thought it would do more stuff. Lemme change that.

File: services/pane.js (hunk 4 above)
Comment (yoshokatana) [inline services/pane.js]: Agreed, added.

File: services/pane.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+function addPreview(preview) {
+  if (preview) {
+    return `<span class="error-preview">${preview}</span>`;
+  } else {
+    return '';
+  }
+}
+
+function addErrors(errors) {
```
Comment (cruzanmo) [inline services/pane.js]: Looks like errors array requires objects with certain properties -- need to add to JSDoc.

Comment (yoshokatana) [inline services/pane.js]: Agreed.

**Comment by yoshokatana** [comment]
:+1:  from dane

**Pull request merged**